tests/Unit/ActionTesting.hpp

```diff
@@ -926,4 +949,66 @@ class MockRuntimeSystem {
   Inboxes inboxes_;
   TupleOfMockDistributedObjects local_algorithms_;
 };
+
+
+/// Returns a vector of all the indices of the Components
+/// in the ComponentList that have queued actions.
+template <typename ComponentList, typename MockRuntimeSystem>
+std::vector<size_t> indices_of_components_with_queued_actions(
+    MockRuntimeSystem& runner) noexcept {
+  std::vector<size_t> result{};
+  size_t i = 0;
+  tmpl::for_each<ComponentList>([&](auto tag) noexcept {
+    using Tag = typename decltype(tag)::type;
+    if (not runner.template is_simple_action_queue_empty<Tag>(0)) {
+      result.push_back(i);
+    }
+    ++i;
+  });
+  return result;
+}
+
+/// \cond
+namespace detail {
+// Helper struct that invokes the next queued action on
+// the ith Component.
+template <typename ComponentList, typename MockRuntimeSystem,
```

Comment:
`MockRuntimeSystem` can be a template parameter of the function, then it can be inferred in the using code.